File: pythonproject/service/estatistica_demandas.py
from service.estatistica_base import EstatisticaBase
from model.Pessoa import Pessoa
from typing import List

class EstatisticaDemandas(EstatisticaBase):
    CATEGORIAS = [
        "AC", "LB_PPI", "LB_Q", "LB_PCD", "LB_EP",
        "LI_PPI", "LI_Q", "LI_PCD", "LI_EP", "V"
    ]

    def calcular(self) -> List[float]:
        return self.calcular_porcentagens(self.CATEGORIAS, Pessoa.get_tipo_vaga)

File: javaproject/src/main/java/service/EstatisticaDemandas.java
package service;

import model.*;
import java.util.List;

public class EstatisticaDemandas extends EstatisticaBase {
    private static final String[] CATEGORIAS = {
        "AC", "LB_PPI", "LB_Q", "LB_PCD", "LB_EP",
        "LI_PPI", "LI_Q", "LI_PCD", "LI_EP", "V"
    };

    @Override
    public List<Double> calcular() {
        return calcularPorcentagens(CATEGORIAS, Pessoa::getTipoVaga);
    }
}